pkg/resources/provisioning.cattle.io/v1/cluster/validator.go

```diff
@@ -97,6 +98,11 @@ func (p *provisioningAdmitter) Admit(request *admission.Request) (*admissionv1.A
 		return nil, err
 	}
 
+	if request.Operation == admissionv1.Delete && oldCluster.Name == localCluster {
+		// deleting "local" cluster could corrupt the cluster Rancher is deployed in
+		return admission.ResponseBadRequest("can't delete local cluster"), nil
```

Comment:
non-blocking nit:  "can't delete local cluster" isn't maybe as informative as it could be.  Maybe also include why in the error since I think this could be displayed to the user.  Or maybe change "can't" to "refusing".

Reply:
How about copy-pasting what k8s complains when we try to delete `kube-system`:
```
Error from server (Forbidden): namespaces "kube-system" is forbidden: this namespace may not be deleted
```

Reply:
Of course, modify the message for cluster resource instead of namespace.